### ISSUE TITLE
zigbee status update dies every few hours

### ISSUE THREAD

**Issue by primalmotion** [zigpy/zigpy-deconz#136]
Hi, 

I have some issues with Home Assistant 0.116.4 installed on a Raspberry Pi 3 B+ running HassOS.

I'm using a Rasbee II shield and everything works smooth until at some point, every few hours, the entire Zigbee network stops working correctly. The commands actually work. If I ask to turn on a light or a smart plug, it always works. However the state of the devices is never updated. If the device was on, turning it off will actually turn off the physical device, but when I reload the HA page, the state is back to on (and vice versa) (which breaks basically everything relying on the state)

![Oct-21-2020 11-53-55](https://user-images.githubusercontent.com/208711/96769613-37f7b400-1394-11eb-971d-30e1ae26340f.gif)

A full HA Core restart always fixes the issue for a little while, until it comes back.

Here are the system info

| key | value |
| :------------- | :---------- |
|arch | armv7l|
|chassis | embedded|
|dev | false|
|docker | true|
|docker_version | 19.03.11|
|hassio | true|
|host_os | HassOS 4.14|
|installation_type | Home Assistant OS|
|os_name | Linux|
|os_version | 4.19.127-v7|
|python_version | 3.8.5|
|supervisor | 249|
|timezone | America/Los_Angeles|
|version | 0.116.4|
|virtualenv | false|

Here is the full log from the last restart  (with some privacy edit)

[home-assistant.log](https://github.com/zigpy/zigpy-deconz/files/5418218/home-assistant.log)

Here are some extracts I find interesting:

First there is this exception that I believe (but I'm not sure) marks the start of the problem. It is shown only once in the log

```
2020-10-21 11:06:59 DEBUG (MainThread) [zigpy_deconz.api] Command Command.aps_data_indication (1, 1)
2020-10-21 11:06:59 DEBUG (MainThread) [zigpy_deconz.uart] Send: 0x178d000800010001
2020-10-21 11:06:59 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x178d050800010022
2020-10-21 11:06:59 WARNING (MainThread) [zigpy_deconz.api] Failed to deserialize frame: b'178d050800010022'
2020-10-21 11:06:59 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
ValueError: 0 is not a valid ADDRESS_MODE

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.8/site-packages/zigpy_deconz/api.py", line 453, in _aps_data_indication
    r = await self._command(
  File "/usr/local/lib/python3.8/site-packages/zigpy_deconz/api.py", line 304, in _command
    return await asyncio.wait_for(fut, timeout=COMMAND_TIMEOUT)
  File "/usr/local/lib/python3.8/asyncio/tasks.py", line 483, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.8/site-packages/zigpy_deconz/api.py", line 336, in data_received
    data, _ = t.deserialize(data[5:], schema)
  File "/usr/local/lib/python3.8/site-packages/zigpy_deconz/types.py", line 7, in deserialize
    value, data = type_.deserialize(data)
  File "/usr/local/lib/python3.8/site-packages/zigpy_deconz/types.py", line 238, in deserialize
    mode, data = ADDRESS_MODE.deserialize(data)
  File "/usr/local/lib/python3.8/site-packages/zigpy_deconz/types.py", line 45, in deserialize
    r = cls(int.from_bytes(data[: cls._size], byteorder, signed=cls._signed))
  File "/usr/local/lib/python3.8/enum.py", line 309, in __call__
    return cls.__new__(cls, value)
  File "/usr/local/lib/python3.8/enum.py", line 600, in __new__
    raise exc
  File "/usr/local/lib/python3.8/enum.py", line 584, in __new__
    result = cls._missing_(value)
  File "/usr/local/lib/python3.8/enum.py", line 613, in _missing_
    raise ValueError("%r is not a valid %s" % (value, cls.__name__))
ValueError: 0 is not a valid ADDRESS_MODE
```

After that I see that every time I try to interact with a zigbee device, I see some errors in the log:

```
2020-10-21 09:19:05 DEBUG (MainThread) [homeassistant.components.zha.core.channels.base] [0x821F:11:0x0006]: command failed: 'on' args: '()' kwargs '{}' exception: '[0x821f:11:0x0006]: Message send failure'
2020-10-21 09:19:05 DEBUG (MainThread) [homeassistant.components.zha.entity] light.hue_chambre_1: turned on: {'on_off': DeliveryError('[0x821f:11:0x0006]: Message send failure')}
```

The message was actually sent (since, well, the device physically reacts), but for some reasons, HA thinks the command failed.

I was asked to report this bug here by @Adminiuga. I can provide any additional information you need.

**Comment by Adminiuga** [comment]
What firmware version is running on the stick?

**Comment by primalmotion** [comment]
how can I know this?

**Comment by Adminiuga** [comment]
You are on `0x26610700` version, based on `2020-10-21 08:49:38 DEBUG (MainThread) [zigpy_deconz.api] Version response: 26610700`

But are you sure you don't have the deCONZ add-on running at the same time with ZHA??? disable and uninstall deCONZ addon, if using ZHA.

You get this in your log
```
2020-10-21 11:06:58 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x0e010007008100
2020-10-21 11:06:58 DEBUG (MainThread) [zigpy_deconz.api] Device state changed response: [<DeviceState.128|1: 129>, 0]
2020-10-21 11:06:58 DEBUG (MainThread) [zigpy_deconz.api] Network state transition: CONNECTED -> JOINING
2020-10-21 11:06:58 DEBUG (MainThread) [zigpy_deconz.api] Data request queue full.
2020-10-21 11:06:58 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x0e02000700a200
2020-10-21 11:06:58 DEBUG (MainThread) [zigpy_deconz.api] Device state changed response: [<DeviceState.128|APSDE_DATA_REQUEST_SLOTS_AVAILABLE|2: 162>, 0]
2020-10-21 11:06:58 DEBUG (MainThread) [zigpy_deconz.api] Network state transition: JOINING -> CONNECTED
2020-10-21 11:06:59 WARNING (MainThread) [zigpy_deconz.api] No response to 'Command.aps_data_confirm' command with seq id '0x8c'
```

And that should never happen, unless some other app is tryung to access and control the same stick.

**Comment by Adminiuga** [comment]
the transition from "connected -> joining" and from "joining -> Connected"

**Comment by primalmotion** [comment]
I don't think I have it installed 

<img width="1177" alt="Screen Shot 2020-10-21 at 13 45 14" src="https://user-images.githubusercontent.com/208711/96785564-d3445580-13a3-11eb-977b-fa8f745465e1.png">

```
docker ps
CONTAINER ID        IMAGE                                              COMMAND                  CREATED             STATUS              PORTS                  NAMES
ab46fe914ca8        hassioaddons/ssh-armv7:7.6.0                       "/init"                  11 minutes ago      Up 11 minutes                              addon_a0d7b954_ssh
ec3326c0e603        homeassistant/armv7-hassio-supervisor:latest       "/init"                  6 hours ago         Up 6 hours                                 hassio_supervisor
36249addb6f9        homeassistant/armv7-addon-configurator:5.1.0       "/init"                  2 days ago          Up 2 days                                  addon_core_configurator
4a97fbeb58d4        homeassistant/armv7-hassio-cli:2020.10.0           "/init /bin/bash -c …"   3 days ago          Up 3 days                                  hassio_cli
d5f20f5e293c        homeassistant/armv7-hassio-multicast:3             "/init"                  3 days ago          Up 6 hours                                 hassio_multicast
671c8dfcb0a8        homeassistant/armv7-hassio-audio:17                "/init"                  3 days ago          Up 3 days                                  hassio_audio
34c2a9abbea9        homeassistant/armv7-hassio-dns:9                   "/init coredns -conf…"   3 days ago          Up 6 hours                                 hassio_dns
3986dc4274b7        homeassistant/raspberrypi3-homeassistant:0.116.4   "/init"                  4 days ago          Up 5 hours                                 homeassistant
70dcd2f06b9a        homeassistant/armv7-hassio-observer:2020.10.1      "/init"                  4 days ago          Up 3 days           0.0.0.0:4357->80/tcp   hassio_observer
```

```
ps -ef | grep -i deconz
 1324 root      0:00 grep -i deconz
```

**Comment by Adminiuga** [comment]
How did you update the ConBee to the latest version? I wonder if you should try the previous version.

**Comment by primalmotion** [comment]
What is the ConBee? If you mean the Raspbee II shield, I just bought it a few days ago. It's brand new

**Comment by Adminiuga** [comment]
Well, Raspee II and ConBee II they both have firmware version ending in 0700. I have no way of knowing it otherwise. 
Interesting, cause that firmware was release on 10/7. Was not expecting them to ship with the latest firmware that fast.

I need to fix the issue, so it doesn't throw the exception, but at the same time you still going to see some issues, cause there was no response to a command which was sent to conbee, just before it went  through "Connected -> Joining -> Connected" states. Another reason for the behavior, if the stick resets internally.

**Comment by primalmotion** [comment]
So you suggest I go on an adventure and downgrade the firmware? If so what version do you suggest?

**Comment by Adminiuga** [comment]
Actually, in approx 3 hours you got 27 spontaneous disconnects: 

```
2020-10-21 10:53:16 DEBUG (MainThread) [zigpy_deconz.api] Device state changed response: [<DeviceState.128|1: 129>, 0]
2020-10-21 10:53:16 DEBUG (MainThread) [zigpy_deconz.api] Network state transition: CONNECTED -> JOINING
2020-10-21 10:53:16 DEBUG (MainThread) [zigpy_deconz.api] Data request queue full.
2020-10-21 10:53:17 DEBUG (MainThread) [homeassistant.components.zha.entity] light.hue_chambre_1: polling current state - from cache: False
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.zigbee.application] Sending Zigbee request with tsn 105 under 106 request id, data: b'0069000000'
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] Command Command.aps_data_request (20, 106, 0, <DeconzAddressEndpoint address_mode=2 address=0x821F endpoint=11>, 260, 6, 1, b'\x00i\x00\x00\x00', 2, 0)
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.uart] Send: 0x129a001b0014006a00021f820b0401060001050000690000000200
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x129a0609000200016a
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] 'aps_data_request' failure: Command.aps_data_request, status: Status.NO_NETWORK
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy.device] [0x821f] Delivery error for seq # 0x69, on endpoint id 11 cluster 0x0006: Couldn't enqueue send data request: Command.aps_data_request, status: Status.NO_NETWORK
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.zigbee.application] Sending Zigbee request with tsn 107 under 108 request id, data: b'006b000000'
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] Command Command.aps_data_request (20, 108, 0, <DeconzAddressEndpoint address_mode=2 address=0x821F endpoint=11>, 260, 8, 1, b'\x00k\x00\x00\x00', 2, 0)
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.uart] Send: 0x129b001b0014006c00021f820b04010800010500006b0000000200
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x129b0609000200016c
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] 'aps_data_request' failure: Command.aps_data_request, status: Status.NO_NETWORK
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy.device] [0x821f] Delivery error for seq # 0x6b, on endpoint id 11 cluster 0x0008: Couldn't enqueue send data request: Command.aps_data_request, status: Status.NO_NETWORK
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.zigbee.application] Sending Zigbee request with tsn 109 under 110 request id, data: b'006d000700030004000240'
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] Command Command.aps_data_request (26, 110, 0, <DeconzAddressEndpoint address_mode=2 address=0x821F endpoint=11>, 260, 768, 1, b'\x00m\x00\x07\x00\x03\x00\x04\x00\x02@', 2, 0)
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.uart] Send: 0x129c0021001a006e00021f820b04010003010b00006d0007000300040002400200
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x129c0609000200016e
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] 'aps_data_request' failure: Command.aps_data_request, status: Status.NO_NETWORK
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy.device] [0x821f] Delivery error for seq # 0x6d, on endpoint id 11 cluster 0x0300: Couldn't enqueue send data request: Command.aps_data_request, status: Status.NO_NETWORK
2020-10-21 10:53:17 DEBUG (MainThread) [homeassistant.components.zha.core.channels.base] [0x821F:11:0x0300]: failed to get attributes '['color_temperature', 'current_x', 'current_y', 'color_loop_active']' on 'light_color' cluster: [0x821f:11:0x0300]: Message send failure
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x0e9d000700a200
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] Device state changed response: [<DeviceState.128|APSDE_DATA_REQUEST_SLOTS_AVAILABLE|2: 162>, 0]
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] Network state transition: JOINING -> CONNECTED
2020-10-21 10:53:22 DEBUG (MainThread) [homeassistant.components.zha.entity] light.hue_living_color_1: polling current state - from cache: False
```

While it is "not connected", it can't send anything to the network, which makes sense
```
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x129a0609000200016a
2020-10-21 10:53:17 DEBUG (MainThread) [zigpy_deconz.api] 'aps_data_request' failure: Command.aps_data_request, status: Status.NO_NETWORK
```

I'm pretty sure that's not a normal behavior of a RaspBee II adapter. Only seen those transition when forming network, happens only once. 
I don't know if it is a hardware issue or a software one.

**Comment by Adminiuga** [comment]
You could try a previous firmware version http://deconz.dresden-elektronik.de/deconz-firmware/deCONZ_RaspBeeII_0x26650700.bin.GCF but updating on a docker would require a container from marthoc https://github.com/marthoc/docker-deconz#updating-conbeeraspbee-firmware

**Comment by Adminiuga** [comment]
But with so many disconnects I'd not expect it to function normally, cause it looks like after each disconnect it takes a bit for the network to recover

**Comment by primalmotion** [comment]
I have downgraded the firmware to deCONZ_RaspBeeII_0x26610700.bin.GCF from 08-Jun-2020 (smallest version available from http://deconz.dresden-elektronik.de/deconz-firmware/) and restarted my stuff. I'll see how it goes. I wanted to get the change log, but I can't find it anywhere

**Comment by Adminiuga** [comment]
The one I've linked is a bit newer, from Sep 12th 2020
in about 1 hour check the log for `CONNECTED -> JOINING` hits. in your previous log, the longs interval between getting that message was less than 30min

**Comment by primalmotion** [comment]
Ok. I started this very test but with the version I installed. I will let it run for a while, see if it works, and then upgrade to latest and test again, see if `26610700` is actually the culprit.

**Comment by primalmotion** [comment]
so `26610700` is also reconnecting every 10/20m. Now retrying using firmware `26650700`

**Comment by Adminiuga** [comment]
Hrm, then it is not a firmware issue.

**Comment by Adminiuga** [comment]
Try running it with deCONZ

**Comment by Adminiuga** [comment]
so if you want to give it a try, here's a "drop in" test version which should fix the issue of zigpy_deconz stop blowing up when adapter behaves un-expectedly.
To install, run in HASS developer shell:

```
docker exec -ti homeassistant pip3 uninstall -y zigpy_deconz
docker exec -ti homeassistant pip3 install https://codeload.github.com/Adminiuga/zigpy-deconz/zip/fixes/error-deserialize-drop-in
```
and restart home assistant. If everything is correct, then HA should not try to re-install the another version of zigpy-deconz

**Comment by Adminiuga** [comment]
give it a try. it won't fix all the problem -- as disconnects are outside its control, but at least while connected you should be able to control the lights. 

Get the logs if you give it a try. If it is better, i'll publish a new version to be included in the next release.

**Comment by primalmotion** [comment]
I will give it a go. I blew up my entire network a few hours ago so I just reset everything when I realized that my hue bridge integration was still somehow active (I have no idea how or why). I basically nuked all trace of all zigbee related files and re-paired everything but the hues. It's been a few hours now and not a single reconnect in the log.

Tomorrow I will try to repair one hue and see if it eventually fails after a few days

**Comment by MattWestb** [comment]
@Adminiuga I was thinking of PAN ID, Extended-PAN ID and channel conflict or WiFi interference but i don't knowing how Rasp/CornBee is handle that and if getting some error codes from the NCP.
Sounds strange that the NCP is restarting / leveling / rejoining, it must being one reason for that can being one HW problem but if you also getting it but not so frequent its not very likely.

@primalmotion Check what channel your HUE GW is using and force ZHA using one other channel  (11, 15, 20 and 25 is primary ones and is the best to using) and if possible looking for some zigbee channel with less WiFi signals straight for eliminating interference.

**Comment by primalmotion** [comment]
So shit started to go south again. It was stable with a few hues, but then today I added the 3 remaining ones, the disconnection started again. I installed your change now, I'll report if it fails the exception killing zha at least

**Comment by primalmotion** [comment]
@Adminiuga  Ok so your fix works. I still have this JOIN/CONNECTED flap every 10m or so now, but the ZHA doesn't die at least

**Comment by primalmotion** [comment]
@MattWestb So I figured out how to change that. I've followed this guide here https://www.metageek.com/training/resources/zigbee-wifi-coexistence.html and switch my 2.5Ghz wifi to channel 6 (previously 11) and set ZHA to use channel 25 (previously ??). It's been 30m without a flap. Finger crossed...

**Comment by MattWestb** [comment]
If you have one mobile use one wifi scanner and you can see if you have some strong wifi around. Normally is the strongest your own WiFi so not a big problems if you have nice neburs that is not doing heavy up/downloads all the time ;-)

I think ZHA is using 15 as standard if not putting something in the config.

WiFi 6 and Zigbee 25 sud being one of the best setups.

I hope some more hours without "Flaps".

**Comment by Adminiuga** [comment]
when HA restarts, look for the following messages
```
2020-10-21 08:49:41 DEBUG (MainThread) [zigpy_deconz.api] Command Command.read_parameter (1, <NetworkParameter.current_channel: 28>, b'')
2020-10-21 08:49:41 DEBUG (MainThread) [zigpy_deconz.uart] Send: 0x0a2500080001001c
2020-10-21 08:49:41 DEBUG (MainThread) [zigpy_deconz.uart] Frame received: 0x0a2500090002001c19
2020-10-21 08:49:41 DEBUG (MainThread) [zigpy_deconz.api] Read parameter current_channel response: [25]
```

per your old log you are on channel 25

**Comment by primalmotion** [comment]
So it's been almost an hour without a flap. I have a pretty powerful wifi router that is a like 50cm away from the Raspbee as well as a repetitor. Neighboor WiFi networks are negligeable. It seems that moving the wifi on channel 6 is doing the trick. I'll declare success this evening :)

**Comment by Adminiuga** [comment]
The fix will be in HA 0.117.0  Or in beta, starting with 0.117.0b2 version

**Comment by primalmotion** [comment]
Thanks! I still see some disconnect but they are not pretty rare (2 in the last 3h). I guess I can live with it for now and we can close this issue.

Just wanted to give you and all HA teams a shout out. This project is marvelous. I used to use SmartThings, Hue, Harmony, Nanoleaf etc. SmartThings was very cool a few years ago but since Samsung bought it, it's going more and more down the toilet. Also binding all these shit cloud together was a bug nest. 

I'm very happy I found HA which allows me to replace everything, have everything run truly locally and provides very VERY nice interface and static config.

Thanks guys!

**Comment by MattWestb** [comment]
RaspBee dont like network manager then its trying "pulling" all devices that can acting as one network adapter also comports. DE is recommending uninstalling it if its installed in the distro.
Is some more things that is written in the install guide (on Rpi disabling or moving BT to soft commport so NCP can using the hardware comport disabling local consolle that using the commport by default).
